discovery(schemas): UsageRecordRequest collision in schemas_analytics.py — Phase 35 shadows LLM-usage variant

## Problem

`autobot-backend/api/schemas_analytics.py` has TWO `class UsageRecordRequest(BaseModel)` definitions:

- **Line 2845** — LLM usage analysis (fields: `prompt`, `model`, `input_tokens`, `output_tokens`, `response_time`, `success`, `session_id`; has `to_record_dict()` helper). Migrated from `analytics_llm_patterns.py` or similar.
- **Line 3276** — `/api/usage/record` request body (fields: `provider`, `model`, `input_tokens`, `output_tokens`, `session_id`, `user_id`, `agent_id`, `latency_ms`, `success`). Added by my Phase 35 migration (PR #6601, commit `98d506ace`).

Last-definition-wins, so any caller importing `UsageRecordRequest` from `schemas_analytics` gets the second one. The first definition (with `to_record_dict()` helper for LLM usage tracking) is dead code.

## Caught by

`flake8 --select=F811 api/schemas_analytics.py`:
```
api/schemas_analytics.py:3276:1: F811 redefinition of unused 'UsageRecordRequest' from line 2845
```

## Fix

Rename the second one (the one I added) to `LLMUsageRecordRequest` for clarity, OR rename the first one to `LLMPatternsUsageRequest` (whichever matches its actual caller). Audit `from api.schemas_analytics import UsageRecordRequest` to determine which callers expect which shape.

## Related

- Phase 35 migration: PR #6601
- This same collision pattern was caught earlier in the project for `ConnectionTestRequest` / `ProviderInfo` (#6604, fixed by VCS-prefix rename).

🤖 Filed during #6617 wiring work; pre-existing bug not caused by current PR.

--- a/autobot-backend/api/schemas_analytics.py
+++ b/autobot-backend/api/schemas_analytics.py
@@ -3286,8 +3286,15 @@ class EmbeddingStatsResponse(BaseModel):
 # ---------------------------------------------------------------------------
 
 
-class UsageRecordRequest(BaseModel):
-    """Request body for POST /api/usage/record."""
+class UsageRecordEndpointRequest(BaseModel):
+    """Request body for POST /api/usage/record.
+
+    Renamed from UsageRecordRequest in #6636 to disambiguate from the
+    LLM-analytics UsageRecordRequest at line 2845, which has a different
+    shape (prompt-based) and is used by analytics_llm_patterns.py and
+    llm_interface_pkg/interface.py. The previous shadowing caused silent
+    Pydantic validation failures in LLM usage tracking
+    (logged as non-critical and swallowed)."""
 
     provider: str
     model: str

--- a/autobot-backend/api/usage.py
+++ b/autobot-backend/api/usage.py
@@ -26,7 +26,7 @@ from api.schemas_common import UsageRecordResponse
 from api.schemas_analytics import (
     UsageByUserAllResponse,
     UsageMyUsageResponse,
-    UsageRecordRequest,
+    UsageRecordEndpointRequest,
     UsageByUserSingleResponse,
     UsageSummaryResponse,
 )
@@ -206,7 +206,7 @@ async def get_my_usage(
     error_code_prefix="USAGE",
 )
 async def record_usage_event(
-    body: UsageRecordRequest,
+    body: UsageRecordEndpointRequest,
     current_user: dict = Depends(get_current_user),
 ) -> dict[str, Any]:
     """

